### PR TITLE
REVAMP: Ready for deployment

### PR DESCRIPTION
**Pull request by AndrewGibson27**
#### What's the plan?
I think you know.

#### What's this PR do?
Minus our fun little dots, which I'll code over the weekend, the JavaScript portion of this is finished. That leaves styling for next sprint.

I also updated the README on this branch to explain the new dev and build processes.

You can skip to the manual testing steps now.

#### How should this be manually tested?
1. Get up and running by following the steps in the new README. If you encounter issues, let me know, because that probably means my documentation sucks.
2. `npm run dev'
3. In your browser, you should see a working -- yet incredibly hideous -- carousel form. Play around with changing the frequencies, ranges and amounts. Is everything dynamically updating as outlined in Ben's mockups? When you click the submit button, is it taking you to the checkout page with the proper value and frequency?
4. Check out my JS. `raw/js/form.js` is where all the logic lives. `raw/js/index.js` is the Webpack entry point that initializes the `FormHandler` class. Does the logic make sense? Is it modular enough?
5. Lastly, if you so desire, check out the tests in `tests/`. You can run them with `npm test`. Do they make sense? Anything I should be checking that I'm not?

 The form class is initialized in `raw/js/index.js`.

#### TODOs:
* [ ] your TODO here

### CODE REVIEW

**Comment by brickaa** [comment]
On `npm run build` I'm getting the following error:
```
  Donation carousel form
    1) "before each" hook for "inner container should be outer container width times number of slides"


  0 passing (2s)
  1 failing

  1) Donation carousel form "before each" hook for "inner container should be outer container width times number of slides":
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

Overall, this looks really great though. Before moving on, can you update the Readme with the new build process and distinguish how you're separating old templates/scripts/etc from the new stuff? It will help keep it organized for when we later refactor or update the circle and other support pages.

**Comment by AndrewGibson27** [comment]
@brickaa Docs updated, and build process fixed!

File: README.md [hunk 1]
```diff
@@ -1,62 +1,78 @@
-# donations-app
-
 ## Getting Started
+To get started:
+1. Clone down the project repo
+2. Make sure [rbenv](https://github.com/sstephenson/rbenv) is installed on your machine
+3. Download Ruby 2.3.0, the version used in this project: `rbenv install 2.0.0`
```
Comment (brickaa) [inline README.md:5]: Should this be `rbenv install 2.3.0`?

File: raw/js/form.js [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+
+  // sets width of inner container as
+  // number of slides * width of outer container
+  setInnerContainerWidth() {
+    const innerContainerWidth = this.numSlides * this.getOuterContainerWidth();
+    this.innerContainer.css('width', `${innerContainerWidth}px`);
+  }
+
+  // returns integer width of inner container
+  getInnerContainerWidth() {
```
Comment (brickaa) [inline raw/js/form.js:48]: Is this used?

**Review by brickaa — COMMENTED** [review]
This looks really good! The `form.js` file is pretty long; I got a bit disoriented trying to follow the smaller functions to where they were used; in a few instances it seems like a function isn't necessary because its only used once and it's easier to know what you're doing if you just see a little extra scripting inline. For example, `isValidAmountInput()` is only used in one place, and it'd be pretty simple to see what you're checking in that if statement, rather than putting in the function. That's more of a personal coding preference though, so nbd. Couple other notes:

+ The whole carousel flashes on load; this isn't a problem at this stage, but you may want to include styles to hide elements until revealed so that they don't flash.
+ On the last slide, there's still a "Next" button
+ Error messaging for string input didn't pop up for me, although I see that's coming along in the script
+ Carousel flashes other slides on resize; we'd probably want the container around the carousel to be 100% width, not a pixel width, if possible, although that may still have issues with the typical carousel technique of absolute positioning things offscreen. Perhaps an alternative is to fake a carousel by using `display: none;` until shown and use CSS transitions/animations to pull it in. Just an idea.

Also, only 1 test ran for me and it failed. IDK what's up with that 
```
  Donation carousel form
    1) "before each" hook for "inner container should be outer container width times number of slides"


  0 passing (2s)
  1 failing

  1) Donation carousel form "before each" hook for "inner container should be outer container width times number of slides":
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
  ```

**Comment by AndrewGibson27** [comment]
@brickaa Thanks for the feedback! 

The one-line functions are more for tests than anything, but I think I can probably clean up a few.

Most of the other issues are expected -- they are things I was planning to address when I tackle the CSS portion of this (next sprint). I was planning to put some kind of loading overlay/hide it until it's ready, and the container won't be a set pixel width -- that was just to test the JS. The next button is getting a disabled class on the last slide -- I just haven't written that class yet. Similar for the error message, when has a hidden class unless input is invalid, just hasn't been styled yet.

Not sure why the tests aren't running. Not crucial to figure out now, but I'll take a look at your screen sometime to see what's happening.

**Comment by AndrewGibson27** [comment]
For the record, @brickaa signed off on this given we've done extensive browser testing. So it's ready for deployment.

**Pull request merged**